src/twisted/internet/defer.py

```diff
@@ -927,6 +928,48 @@ def main(reactor):
     return coro
 
 
+def coroFnToDeferredFn(corofn):
+    """
+    L{coroFnToDeferredFn} helps you define coroutinefunctions where previously
+    there were functions that returned Deferreds
+
+    Coroutine functions return a coroutine object, similar to how generators
+    work. This function turns that coroutinefunction into a Deferred function,
+    meaning that it can be used in legacy Twisted code. For example::
+
+        import treq
+        from twisted.internet import defer, task
+
+        async def crawl(pages):
+            results = {}
+            for page in pages:
+                results[page] = await treq.content(await treq.get(page))
+            return results
+
+        @defer.coroFnToDeferredFn
+        async def main(reactor):
+            pages = [
+                "http://localhost:8080"
+            ]
+            print(await crawl(pages))
+
+        task.react(main)
+
+    @param corofn: The coroutinefunction to wrap
+    @type coroutinefunction: A Python 3.5+ C{async def} C{coroutinefunction}
```

Comment:
Since Twisted is supported only on Python 3.5 and up, I don't think that we need to mention Python3.5 here

Reply:
This is copied from the ensureDeferred docstring, and there's many many legacy mentions of old pythons. I think they should all be handled in one go
```suggestion
    @type coroutinefunction: A C{async def} C{coroutinefunction}
```